SharedCode/src/commonMain/kotlin/ApplicationPresenter.kt

```diff
@@ -67,11 +65,39 @@ class ApplicationPresenter: ApplicationContract.Presenter() {
     }
 
     override fun setDepartureStation(departureStation: String) {
-        chosenDepartureStation = departureStation
+        chosenDepartureStation = matchStationNameToCode(departureStation)
     }
 
     override fun setArrivalStation(arrivalStation: String) {
-        chosenArrivalStation = arrivalStation
+        chosenArrivalStation = matchStationNameToCode(arrivalStation)
+    }
+
+    private fun buildStationDropdowns() {
+        launch {
+            val allStationDetails = getAllStationDetails()
+            buildStationNamesList(allStationDetails)
+            view!!.setDepartureDropdown(stationNames)
+            view!!.setArrivalDropdown(stationNames)
+        }
+    }
+
+    private fun buildStationNamesList(allStationDetails: StationDetails) {
+        for ((codeIndex, stationCode) in stationCodes.withIndex()) {
+            for (stationInformation in allStationDetails.stations) {
+                if (stationInformation.crs == stationCode) {
+                    stationNames[codeIndex] = stationInformation.name
+                }
+            }
+        }
+    }
+
+    private suspend fun getAllStationDetails(): StationDetails {
+        val apiCall = "https://mobile-api-dev.lner.co.uk/v1/stations"
```

Comment:
Not a big deal, but you're now making two calls to the api, so using the 'https://mobile-api-dev.lner.co.uk/v1' base url in two places, do you think it's worth extracting it?

Reply:
Yep was thinking this - I'll do some refactoring